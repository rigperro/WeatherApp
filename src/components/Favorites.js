import React, { useState, useEffect } from 'react';
import styled from 'styled-components';

const Favorites = ({getWeather, setError, city}) => {
  const [favorites, setFavorites] = useState(
    JSON.parse(localStorage.getItem('favorites')) || []
  );

  useEffect(() => {
    localStorage.setItem('favorites', JSON.stringify(favorites));
  }, [favorites]);

  const handleSet = () => {
    if (city) {
      var arr = [...favorites];
      if (favorites.length > 2) {
        arr.splice(2, 1);
      }
      arr.unshift(city);
      setFavorites(arr);
    } else {
      setError("Error: no city chosen")
    }
  }

  return(
    <StyledDiv>
      <StyledButton onClick={handleSet}>Set as favorite</StyledButton>
      {favorites && <p> Your favorites: </p>}
      {(favorites || []).map((item, index) => (
        <StyledButton key={index} onClick={() => {getWeather(item)}}>{item}</StyledButton>
      ))}
    </StyledDiv>
  );
};

const StyledDiv = styled.div`
  display: flex;
  flex-direction: column;
`

const StyledButton = styled.button`
  background: transparent;
  border-radius: 3px;
  border: 3px solid white;
  color: white;
  margin: 0.5em 1em;
  font-family: 'Oxygen', sans-serif;
  padding: 0.4em 0.8em;
  &:hover {
  background: #260A13;
  }
`

export default Favorites;
