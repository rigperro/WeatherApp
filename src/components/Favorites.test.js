import React from 'react';
import ReactDOM from 'react-dom';
import Favorites from './Favorites';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

it('renders without chrashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Favorites />, div);
});

test('Favorites renders correctly', () => {
  const tree = renderer
    .create(<Favorites />)
    .toJSON();
  expect(tree).toMatchSnapshot();
})
