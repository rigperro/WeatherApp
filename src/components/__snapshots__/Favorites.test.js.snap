// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorites renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c1 {
  background: transparent;
  border-radius: 3px;
  border: 3px solid white;
  color: white;
  margin: 0.5em 1em;
  font-family: 'Oxygen',sans-serif;
  padding: 0.4em 0.8em;
}

.c1:hover {
  background: #260A13;
}

<div
  className="c0"
>
  <button
    className="c1"
    onClick={[Function]}
  >
    Set as favorite
  </button>
  <p>
     Your favorites: 
  </p>
</div>
`;
